python/ray/data/datasource/datasource.py

```diff
@@ -118,6 +118,45 @@ def on_write_failed(
         """
         pass
 
+    @classmethod
+    def _get_datasource_name(cls) -> str:
```

Comment:
Another idea for this: make this a `get_name()` instance method that is public and `Datasource` implementers can override, and make the default implementation something like this:
```python
def get_name(self) -> str:
    """Return a human-readable name for this datasource.

    This will be used as the names of the read tasks.
    """
    name = type(self).__name__
    datasource_suffix = "Datasource"
    if name.endswith(datasource_suffix):
        name = name[:-len(datasource_suffix)]
    return f"Read{name}"
```

We'd only need to override this for `ParquetBaseDatasource` and maybe `RandomIntRowDatasource`, and this would provide a cleaner path for custom datasources to have a nice name.